homeassistant/components/homekit/type_fans.py

```diff
@@ -0,0 +1,159 @@
+"""Class to hold all light accessories."""
+import logging
+
+from pyhap.const import CATEGORY_FAN
+
+from homeassistant.components.fan import (
+    ATTR_DIRECTION, ATTR_OSCILLATING, ATTR_SPEED, ATTR_SPEED_LIST,
+    DIRECTION_FORWARD, DIRECTION_REVERSE, SUPPORT_DIRECTION,
+    SUPPORT_OSCILLATE, SUPPORT_SET_SPEED)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_SUPPORTED_FEATURES, STATE_OFF, STATE_ON,
+    SERVICE_TURN_OFF, SERVICE_TURN_ON)
+from homeassistant.core import split_entity_id
+
+from . import TYPES
+from .accessories import HomeAccessory, debounce
+from .const import (
+    CHAR_ACTIVE, CHAR_ROTATION_DIRECTION, CHAR_ROTATION_SPEED,
+    CHAR_SWING_MODE, SERV_FANV2)
+
+_LOGGER = logging.getLogger(__name__)
+
+HASS_TO_HOMEKIT = {DIRECTION_FORWARD: 0,
+                   DIRECTION_REVERSE: 1}
+HOMEKIT_TO_HASS = {c: s for s, c in HASS_TO_HOMEKIT.items()}
+
+
+@TYPES.register('Fan')
+class Fan(HomeAccessory):
+    """Generate a Fan accessory for a fan entity.
+
+    Currently supports: state, speed, oscillate, direction.
+    """
+
+    def __init__(self, *args, config):
+        """Initialize a new Light accessory object."""
+        super().__init__(*args, category=CATEGORY_FAN)
+        self._domain = split_entity_id(self.entity_id)[0]
+        self._flag = {CHAR_ACTIVE: False,
+                      CHAR_ROTATION_SPEED: False,
+                      CHAR_ROTATION_DIRECTION: False,
+                      CHAR_SWING_MODE: False}
+        self._state = 0
+
+        self.chars = []
+        self._features = self.hass.states.get(self.entity_id) \
+            .attributes.get(ATTR_SUPPORTED_FEATURES)
+        if self._features & SUPPORT_SET_SPEED:
+            self.chars.append(CHAR_ROTATION_SPEED)
+            self._speed = None
```

Comment:
What reason is their to define it as class variable? As I see it, it's only used in `update_state`. Same for `self._direction` and `self._oscillating`.

Reply:
Sorry, I just copied the base code from `type_lights`. It's not needed so I removed.